src/main/java/core/basesyntax/strategy/operation/PurchaseHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void getOperation(FruitTransaction transaction) {
+        if (Storage.storage.get(transaction.getFruit()) - transaction.getQuantity() > 0) {
+            int valueBeforeOperation = Storage.storage.get(transaction.getFruit());
+            Storage.storage
+                    .put(transaction.getFruit(), valueBeforeOperation - transaction.getQuantity());
+        } else {
+            throw new RuntimeException("Balance after purchase will be negative");
+        }

```

Comment:
it's better to refactor if-else throw constructions to if-throw ones, inverting the logical condition. We don't need else after throw, and returning early is a preferred approach, because nested code is harder to read